Zend/zend_compile.c

```diff
@@ -6284,6 +6309,14 @@ void zend_compile_class_decl(zend_ast *ast, zend_bool toplevel) /* {{{ */
 	}
 
 	CG(active_class_entry) = ce;
+	if (extends_ast) {
+		zend_string *extends_name = zend_ast_get_str(extends_ast);
+		CG(active_class_entry_parent_name) = zend_resolve_class_name(
+			extends_name,
+			extends_ast->kind == ZEND_AST_ZVAL ? extends_ast->attr : ZEND_NAME_FQ);
```

Comment:
This is no longer necessary, you can use `ce->parent_name`.

Reply:
Thanks - will rework the patch.

Reply:
@nikic How do I know if it's safe to use parent or parent_name in that union?

Reply:
During compilation it should always be parent_name. Otherwise the ZEND_ACC_LINKED flag.